src/lambdas/ListFiles/test/index.jest.ts

```diff
@@ -0,0 +1,102 @@
+}
+
+const batchGetMock = jest.fn()
+const queryMock = jest.fn()
+jest.unstable_mockModule('../../../lib/vendor/AWS/DynamoDB', () => ({
+  batchGet: batchGetMock,
+  query: queryMock
+}))
+
+const {handler} = await import('./../src')
+
+describe('#ListFiles', () => {
+  const context = testContext
+  let event: CustomAPIGatewayRequestAuthorizerEvent
+  beforeEach(() => {
+    event = JSON.parse(JSON.stringify(eventMock))
+    process.env.DynamoDBTableFiles = 'Files'
+    process.env.DynamoDBTableUserFiles = 'UserFiles'
+  })
+  test('(anonymous) should list only the default file', async () => {
+    delete event.headers['X-User-Id']
+    delete event.headers['Authorization']
+    const {default: batchGetResponse} = await import('./fixtures/batchGet-200-OK.json', {assert: {type: 'json'}})
+    batchGetMock.mockReturnValue(batchGetResponse)
+    queryMock.mockReturnValue(queryStubReturnObject)
+    const output = await handler(event, context)
+    expect(output.statusCode).toEqual(200)
+    const body = JSON.parse(output.body)
+    expect(Object.keys(body.body)).toEqual(expect.arrayContaining(['keyCount', 'contents']))
+    expect(body.body.keyCount).toEqual(1)
+    expect(body.body.contents[0]).toHaveProperty('authorName')
+    expect(body.body.contents[0].authorName).toEqual('Lifegames')
+  })
+  test('(authenticated) should return users files', async () => {
+    event.requestContext.authorizer!.principalId = fakeUserId
+    const {default: batchGetResponse} = await import('./fixtures/batchGet-200-Filtered.json', {assert: {type: 'json'}})
+    batchGetMock.mockReturnValue(batchGetResponse)
+    queryMock.mockReturnValue(queryStubReturnObject)
+    const output = await handler(event, context)
+    expect(output.statusCode).toEqual(200)
+    const body = JSON.parse(output.body)
+    expect(Object.keys(body.body)).toEqual(expect.arrayContaining(['keyCount', 'contents']))
+    expect(body.body.keyCount).toEqual(1)
+  })
+  test('(authenticated) should gracefully handle an empty list', async () => {
+    event.requestContext.authorizer!.principalId = fakeUserId
+    const {default: batchGetResponse} = await import('./fixtures/batchGet-200-Empty.json', {assert: {type: 'json'}})
+    batchGetMock.mockReturnValue(batchGetResponse)
+    const {default: queryResponse} = await import('./fixtures/query-200-Empty.json', {assert: {type: 'json'}})
+    queryMock.mockReturnValue(queryResponse)
+    const output = await handler(event, context)
+    expect(output.statusCode).toEqual(200)
+    const body = JSON.parse(output.body)
+    expect(Object.keys(body.body)).toEqual(expect.arrayContaining(['keyCount', 'contents']))
+    expect(body.body.keyCount).toEqual(0)
+  })
+  test('should fail gracefully if query fails', async () => {
+    queryMock.mockImplementation(() => {
+      throw new Error()
+    })
+    const output = await handler(event, context)
+    expect(output.statusCode).toEqual(401)
```

Comment:
why is it a 4XX response when the server had an internal issue with the query here? i would think that should be a 5XX of some kind here. but maybe i'm missing something?